ti/config/config.go

```diff
@@ -127,3 +128,14 @@ func (c *Cfg) GetFeatureState(stepID string, feature types.SavingsFeature) (type
 	}
 	return types.DISABLED, ErrStateNotFound
 }
+
+func (c *Cfg) LockZipForRuby() {
```

Comment:
Access to this variable rubylocked should be done using a mutex since multiple steps will be reading and writing to it at the same time

Reply:
Anyway I had to use an extra variable to check the status of lock, as in go there is no direct way to check the status of lock so added this directly,
I will add a mutex as well